cli/src/lib.rs

```diff
@@ -916,35 +898,24 @@ fn init(
     if solidity {
         solidity_template::create_program(&project_name)?;
     } else {
-        rust_template::create_program(&project_name, template)?;
+        let tests_mod = match test_template {
+            TestTemplate::Jest | TestTemplate::Mocha => None,
+            TestTemplate::Rust => Some("tests"),
+        };
+        rust_template::create_program(&project_name, &template, tests_mod)?;
```

Comment:
Same here, the logic of tests doesn't belong inside program creation logic. We already crate test files in this function so they should be coupled.